google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/MessageDispatcher.java

```diff
@@ -103,7 +109,7 @@ void extendExpiration(Instant now) {
       expiration = now.plus(Duration.standardSeconds(nextExtensionSeconds));
```

Comment:
We should probably take into account the maxAckExtensionPeriod here and not modifyAckDeadline beyond that boundary.